packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -2249,31 +2249,36 @@ class _InputDecoratorState extends State<InputDecorator> with TickerProviderStat
     final String? hintText = decoration.hintText;
     final bool maintainHintHeight = decoration.maintainHintHeight;
     Widget? hint;
-    if (hintText != null) {
+    if (decoration.hint != null || hintText != null) {
+      final Widget hintWidget;
+      hintWidget =
```

Comment:
This should be one line I think.

Reply:
Dart format makes it two liner, I suspect the analyzer test would fail making it one line

Reply:
Justin suggestion was related to merge the declaration and the initialization on the same line:

```suggestion
      final Widget hintWidget =
```

Reply:
I have updated that change